contracts/modules/TransferManager/VolumeRestrictionTransferManager.sol

```diff
@@ -0,0 +1,241 @@
+    function addLockUp(address userAddress, uint lockUpPeriodSeconds, uint releaseFrequencySeconds, uint startTime, uint totalAmount) public withPerm(ADMIN) {
+        
+        // if a startTime of 0 is passed in, then start now.
+        if (startTime == 0) {
+            startTime = now;
+        }
+        
+        // Get an array of the user's lockups
+        LockUp[] storage userLockUps = lockUps[userAddress];
+        
+        // create a new lock up and push it into the array of the user's lock ups
+        LockUp memory newLockUp;
+        newLockUp.lockUpPeriodSeconds = lockUpPeriodSeconds;
+        newLockUp.releaseFrequencySeconds = releaseFrequencySeconds;
+        newLockUp.startTime = startTime;
+        newLockUp.totalAmount = totalAmount;
+        userLockUps.push(newLockUp);
+
+        emit LogModifyLockUp(
+            userAddress,
+            lockUpPeriodSeconds,
+            releaseFrequencySeconds,
+            startTime,
+            totalAmount,
+            LockUpOperationType.Add
+        );
+    }
+
+    // same as addLockup, but takes an array for each parameter
+    function addLockUpMulti(address[] userAddresses, uint[] lockUpPeriodsSeconds, uint[] releaseFrequenciesSeconds, uint[] startTimes, uint[] totalAmounts) public withPerm(ADMIN) {
+        
+        // make sure input params are sane
+        require(
+            userAddresses.length == lockUpPeriodsSeconds.length && 
+            userAddresses.length == releaseFrequenciesSeconds.length &&
+            userAddresses.length == startTimes.length &&
+            userAddresses.length == totalAmounts.length,
+            "Input array length mis-match"
+        );
+
+        for (uint i = 0; i < userAddresses.length; i++) {
+            addLockUp(userAddresses[i], lockUpPeriodsSeconds[i], releaseFrequenciesSeconds[i], startTimes[i], totalAmounts[i]);
+        }
+
+    }
+    
+    // remove a user's lock up
+    function removeLockUp(address userAddress, uint lockUpIndex) public withPerm(ADMIN) {
+        LockUp[] storage userLockUps = lockUps[userAddress];
+        LockUp memory toRemove = userLockUps[lockUpIndex];
+
+        emit LogModifyLockUp(
+            userAddress,
+            toRemove.lockUpPeriodSeconds,
+            toRemove.releaseFrequencySeconds,
+            toRemove.startTime,
+            toRemove.totalAmount,
+            LockUpOperationType.Remove
+        );
+
+        // move the last element in the array into the index that is desired to be removed.  
+        userLockUps[lockUpIndex] = userLockUps[userLockUps.length - 1];
+        // delete the last element
+        userLockUps.length--;
+    }
+
+    function editLockUp(address userAddress, uint lockUpIndex, uint lockUpPeriodSeconds, uint releaseFrequencySeconds, uint startTime, uint totalAmount) public withPerm(ADMIN) {
+        
+        // if a startTime of 0 is passed in, then start now.
+        if (startTime == 0) {
+            startTime = now;
+        }
+        
```

Comment:
sanity checks are missing

Reply:
Hey @satyamakgec - could you clarify this one?  What kind of sanity check were you thinking?

It was my thinking that a startTime before "now" may be desired if you want to backdate someone's vesting, which I think is a common thing.  And obviously a startTime after "now" could be desired if you're awarding tokens to an employee who, for example, starts working in 2 weeks or something.